File: react-frontend/project/src/service/ItemService.js
import axios from 'axios';

const Item_Base_URL = "http://localhost:8080/items";

class ItemService {

    // get all item list
    getItems(){
        return axios.get(Item_Base_URL);
    }

    // save item
    addItem(item){
        return axios.post(Item_Base_URL, item);
    }

    // get item by id
    getItemById(itemID){
        return axios.get(Item_Base_URL + '/' + itemID);
    }

    // get item by name
    getItemByName(itemName) {
        return axios.get(`${Item_Base_URL}/${itemName}`);
    }
      
}

const itemService = new ItemService();
export default itemService;

File: springboot-backend/HostBuddy-RESTapi-CRUD/src/main/java/com/hostbuddy/service/ItemService.java
package com.hostbuddy.service;

import java.util.List;

import com.hostbuddy.dto.Item;

public interface ItemService {
	
//	save item
	public Item addItem(Item item);
	
//	get all items
	List<Item> getAllItems();
	
//	get item by name
	public List<Item> getItemByName(String name);
	
//	update item
	public Item updateItem (Item item, int id);
}